src/components/FunctionSeries.tsx

```diff
@@ -0,0 +1,19 @@
+import { SeriesPoint } from '../types';
+
+import { LineSeries, LineSeriesProps } from './LineSeries';
+
+export interface FunctionSeriesProps extends Omit<LineSeriesProps, 'data'> {
+  data: (x: number) => number;
+  max: number;
+  min: number;
+  step?: number;
```

Comment:
The idea of this series is not to have to enter a step and that it is being calculated automatically depending the min / max and the size of the chart (number of point for the width).

Please could you remove this parameter ?

I'm also confused by the min and max props. In the issue the idea was to use the min / max of the xAxis. If there is no other series from which we could extract the min and max value it means that we need to explicitly defined min and max on the axis (and not on the series).

Finally in order to test it is better if this storybook has the square zoom capability.